### DISCUSSION TITLE
Purpose of the 'dynamicPrompts' property in object_info

### DISCUSSION THREAD

**Discussion by ImDarkTom** [Comfy-Org/ComfyUI#5560] [Q&A] [answered]
I've been developing an app that relies on the ComfyUI API and one of the routes I frequently use is `/api/object_info` to get info on the inputs for various nodes, one property that I sometimes see across text inputs is `dynamicPrompts` that is nearly always set to true. 

Does this property actually do anything? Any reference to it in the code seems to just be setting it to true for various nodes and I want to know if there is anything it does so I can account for it in my app. Any help appreciated.

![image](https://github.com/user-attachments/assets/4eb0ae79-6cd2-432a-9aaf-ed40ac56ce4f)

**Comment by ltdrdata** [comment, marked as answer]
This is a feature used in nodes like CLIPTextEncode where dynamicPrompts being set to true determines whether random selection syntax like `{a|b|c}` should be enabled.

Important note: The responsibility for handling this feature lies with the frontend, so if you're implementing an API program, when dynamicPrompts is set to true, your program needs to handle this preprocessing before sending to /prompt.

**Reply by ImDarkTom** [reply]
Thanks
